broker/thinBroker.go

```diff
@@ -2494,12 +2472,12 @@ func (tb *ThinBroker) updateLastSuccessParameters(success bool, time string, sid
 func (tb *ThinBroker) LDUpdateEntityAttributes(w rest.ResponseWriter, r *rest.Request) {
 	var context []interface{}
 	eid := r.PathParam("eid")
-	if ctype := r.Header.Get("Content-Type"); ctype == "application/json" || ctype == "application/ld+json" { // //naveen : need to correct it. => Corrected
-		entity := tb.ldGetEntity(eid)
-		if entity != nil {
+	if ctype := r.Header.Get("Content-Type"); ctype == "application/json" || ctype == "application/ld+json" {
+		tb.ldEntities_lock.RLock()
+		if _, ok := tb.ldEntities[eid]; ok == true {
+			tb.ldEntities_lock.RUnlock()
 			//Get a resolved object ([]interface object)
 			resolved, err := tb.ExpandAttributePayload(r, context)
-
 			if err != nil {
 				if err.Error() == "EmptyPayload!" {
 					rest.Error(w, "Empty payloads are not allowed in this operation!", 400)
```

Comment:
[Major] Missing "tb.ldEntities_lock.RUnlock()" which can create a deadlock

Reply:
Incorporated in code